.circleci/config.yml

```
@@ -179,47 +179,195 @@ commands:
           paths:
             - << parameters.install_path >>
 
-defaults:
+  # --------------------------------------------------------------------------
+  # Build Commands
+
+  setup_prerelease_commit_hash:
+    steps:
+      - run:
+          name: Store commit hash and prerelease
+          command: |
+            if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" ]; then echo -n > prerelease.txt; else date -u +"nightly.%Y.%-m.%-d" > prerelease.txt; fi
```

Maybe structure it like below to be easy to read :)
```suggestion
            if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" ]
            then
              echo -n > prerelease.txt
            else
              date -u +"nightly.%Y.%-m.%-d" > prerelease.txt
            fi
            echo -n "$CIRCLE_SHA1" > commit_hash.txt
```

---

Also, we could use the same syntax that we are using in other scripts here, i.e.
```sh
if [[ "$CIRCLE_BRANCH" == release || -n "$CIRCLE_TAG" ]]
```

Instead of:
```sh
if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" ]
```

Or the idea is to be POSIX-compliant? Although, I believe we always run it on Bash, no?

---

Refactored.

> Or the idea is to be POSIX-compliant? Although, I believe we always run it on Bash, no?

No, this is just a pretty old bit of code and was written without much consideration to that.

I think we should target Bash, because (at least in CI) we don't have to deal with any platform where it would not be available. And portable shell code is horrible to read :) And very often not really portable if you don't actually test it on platforms with these quirks.